Tools/scripts/pathfix.py

```diff
@@ -188,15 +194,26 @@ def parse_shebang(shebangline):
     return shebangline[start:]
 
 
+def populate_flags(shebangline):
+    old_flags = b''
+    if keep_flags:
+        old_flags = parse_shebang(shebangline)
+        if old_flags:
+            old_flags = old_flags[2:]
+    if not (old_flags or add_flags):
+        return b''
+    return b' -' + add_flags + old_flags
```

Comment:
Please add a comment explinaing why new flags are added before old flags.

I guess that the idea is to add -i to "#!python3 -W default" to get: "#!python3 -iW default".

If I recall correctly, a shebang can only have a single option with an argument (-option arg): "-W default -W ignore" is read as -W "default -W ignore". Please also explain that.

Reply:
Can you please complete your comment to explain why new flags (add_flags) come before old flags (old_flags)? Just say that it's a way to support old_flags which contain whitespaces. Or explain that the Linux kernel pass all options as a single argument to the process: argv[1].